### PR TITLE
fix: new .OwlBot.yaml format allows files to excluded from copy and r…

### PR DESCRIPTION
**Pull request by SurferJeffAtGoogle**
…emove

### CODE REVIEW

**Comment by SurferJeffAtGoogle** [comment]
Example of a PR generated with the new code:
https://github.com/googleapis/nodejs-translate/pull/628

File: packages/owl-bot/test/copy-code.ts [hunk 1]
```diff
@@ -115,9 +119,13 @@ describe('copyDirs', () => {
           source:
             '/google/cloud/asset/.*/.*-java/(grpc-google-cloud-asset-.*)-java',
           dest: '/$1',
-          'rm-dest': '/grpc-google-cloud-asset-.*',
         },
       ],
+      'deep-remove-regex': ['/grpc-google-cloud-asset-.*'],
+      'deep-preserve-regex': [
```
Comment (bcoe) [inline packages/owl-bot/test/copy-code.ts:125]: Were we able to exercise the new logic fully with existing tests? Wondering if we need any additional test cases.

**Review by bcoe — APPROVED** [review]
Just a question about whether we need any additional test cases.

File: packages/owl-bot/test/copy-code.ts (hunk 1 above)
Comment (SurferJeffAtGoogle) [reply packages/owl-bot/test/copy-code.ts:125]: Lines 107 and 108 exercise the new logic, and lines 137 and 146 confirm it behaves as expected.

**Pull request merged**